packages/@coorpacademy-components/src/atom/button-link/index.tsx

```diff
@@ -2,14 +2,45 @@ import React, {useCallback} from 'react';
 import {noop} from 'lodash/fp';
 import classnames from 'classnames';
 import Link from '../link';
+import FaIcon from '../icon';
 import {ICONS} from '../../util/button-icons';
 import propTypes, {ButtonLinkProps, IconType} from './types';
 import style from './style.css';
 
 const getButtonContent = (icon?: IconType, label?: string) => {
-  const {type, position} = icon || {type: '', position: ''};
+  const {type, faIcon, position} = icon || {type: '', position: ''};
+  const {name, color, backgroundColor, size} = faIcon || {
+    name: '',
+    color: '',
+    backGroundColor: '',
+    size: 0
+  };
   const Icon = type && ICONS[type];
 
+  const faIconComponent = (
+    <FaIcon
+      {...{
+        iconName: name,
+        iconColor: color,
+        backgroundColor,
+        size: {
+          faSize: size,
+          wrapperSize: size
+        }
+      }}
+    />
+  );
```

Comment:
this and descructuing should be in the `if (faIcon) {` branch!

```js
if (faIcon) {
 const {name, color, backgroundColor, size} = faIcon
 const faIconComponent = (...)
 return (<div>....)
}
```

related: what is default behavior for `position`? :`left` (according `type` is must be one of these)
(or absence implies the icon does not appear?)

Reply:
> (just an idea, not to apply in practice to keep same structure than last return)
> 
> about the diff, alternative can be to have to two element array, and reverse it if right position
> 
> ```js
> const buttonParts = [ faIconComponent /* could be inlined*/, label ? <span className={style.label}>{label}</span> : null]
> return (
>     <div className={style.buttonContent}>
>         {position === 'right' ? _.reverse(buttonParts) : buttonParts}
>     </div>
> )
> ```
> (`reverse` from `lodash/fp`)

Reply:
Suggestion for rework of the whole function in three step
1. early exist for no icon case
2. forge icon component (depending on icon or faicon)
3. return with proper order of icon and label

```
if (!Icon && !faIcon)  return no icon component

const iconComponent  = faIcon ? <FaIcon iconNam={faIcon.name} ... /> : <Icon className={style.icon} theme="coorpmanager" />
return (
    <div className={style.buttonContent}>
      {position === 'left' ? iconComponent : null}
      {label ? <span className={style.label}>{label}</span> : null}
      {position === 'right' ? iconComponent : null}
    </div>
  );
```